utils/_decorators.py

```diff
@@ -49,20 +50,7 @@ def _ensure_jira_ticket_as_reason(item, reason: str):
         pytest.exit(f"Please set a jira ticket for {nodeid}, instead of reason: {reason}", 1)
 
 
-def _get_skipped_item(item, skip_reason):
-    if inspect.isfunction(item) or inspect.isclass(item):
-        if not hasattr(item, "pytestmark"):
-            item.pytestmark = []
-
-        item.pytestmark.append(pytest.mark.skip(reason=skip_reason))
-
-    else:
-        raise ValueError(f"Unexpected skipped object: {item}")
-
-    return item
-
-
-def _get_expected_failure_item(item, skip_reason, force_skip: bool = False):
+def _get_expected_failure_item(item, skip_reason, force_skip):
```

Comment:
I feel that name is not the best one, this function covers two use case (skipped and expected_failure), so its name should not make us think it cover only one. And the argument `force_skip` adds a little bit to the confusion

What about 

```python
def _add_pytest_marker(item, reason:str, pytest_marker):
   ...

# Or even 

def _get_decorated_item(item, reason:str, pytest_marker):
    ...

# usage for expected failure:
_add_pytest_marker(item, "blabla", pytest.mark.skip if force_skip else pytest.mark.xfail)

# usage for irrelevant and flaky
_add_pytest_marker(item, "blabla", pytest.mark.skip)
```

It would make the intent inside the caller very clear, WDYT ?

Reply:
`def _add_pytest_marker(item, reason:str, pytest_marker):` looks good to me